trl/trainer/grpo_trainer.py

```diff
@@ -1575,6 +1575,8 @@ def _generate_and_score_completions(
             completions = []
             for prompt, completion in zip(prompts, completions_text, strict=True):
                 bootstrap = prompt.pop()["content"] if prompt[-1]["role"] == "assistant" else ""
+                if isinstance(bootstrap, list):
+                    bootstrap = bootstrap[-1].get("text", "")
```

Comment:
let's be a bit more constrained to avoid silent bug

```suggestion
                if isinstance(bootstrap, list):  # for VLM, the bootstrap might have the format [{"type": "text", "text": "..."}]
                    assert len(bootstrap) == 1 and bootstrap[0]["type"] == "text":
                    bootstrap = bootstrap[0]["text"]
```